apps/site/playwright.config.ts

```diff
@@ -10,11 +11,34 @@ export default defineConfig({
   retries: isCI ? 2 : 0,
   workers: isCI ? 1 : undefined,
   reporter: isCI ? [['html'], ['github']] : [['html']],
-  use: {
-    baseURL: process.env.VERCEL_PREVIEW_URL || 'http://127.0.0.1:3000',
-    trace: 'on-first-retry',
-  },
+  ...(() => {
+    const use: PlaywrightTestConfig['use'] = {
+      baseURL: process.env.PLAYWRIGHT_BASE_URL || 'http://127.0.0.1:3000',
+      trace: 'on-first-retry',
+    };
 
+    const webServerCommand = process.env.PLAYWRIGHT_WEB_SERVER_COMMAND;
+    const webServerPort = parseInt(
+      process.env.PLAYWRIGHT_WEB_SERVER_PORT ?? ''
+    );
+    if (webServerCommand && !isNaN(webServerPort)) {
+      use.baseURL = `http://127.0.0.1:${webServerPort}`;
+      return {
+        webServer: {
+          command: webServerCommand,
+          port: webServerPort,
+          stdout: 'pipe',
+          stderr: 'pipe',
+          timeout: 60_000 * 3,
```

Comment:
`port` isn't a property. We should define `baseUrl` outside of this, like I said earlier, and re-use it as `url` here.

Reply:
updated, I am no longer setting `port` 🙂 
(also I didn't mean to commit the `stdout` and `stderr` props I was using them for debugging this locally 😓)